LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/IoTHubDeviceInfo.cs

```diff
@@ -13,7 +15,15 @@ public class IoTHubDeviceInfo
 
         public string GatewayId { get; set; }
 
-        public string NwkSKey { get; set; }
+        [JsonProperty("NwkSKey")]
+        public string NwkSKeyString
+        {
+            get => NwkSKey?.ToString();
+            set => NwkSKey = value is { Length: 32 } some ? LoRaWan.NetworkSessionKey.Parse(some) : null;
```

Comment:
if we set an invalid key ( != 32 length), this will simply ignore it. Should we throw in that case?

Reply:
I think this would never happen in our case, as the serialization on the function side will make sure it's the right format - hence I think it's safe to throw and avoid any other paths to provide an invalid format.

Reply:
Agree - I will change the logic to shortcut if there's an empty string, but always try to parse if the string is non-empty

Reply:
Done in c5f2b47adb25c0d58b985d6cc6a983e3b3450e23